### PR TITLE
Product page template

### PR DESCRIPTION
**Pull request by nienn**
* Merge after #27

<s>_Note: #27 is currently blocked on #23_</s>
_Edit:  #23 has been merged._

### CODE REVIEW

File: readthedocs-theme/templates/readthedocs/product.html
Comment (agjohnson) [inline readthedocs-theme/templates/readthedocs/product.html:9]: If we don't have anything here, it's probably best to remove this section and come back to it later. We want to be moving towards less mockup copy and content in the site.

File: readthedocs-theme/templates/readthedocs/product.html
Comment (agjohnson) [inline readthedocs-theme/templates/readthedocs/product.html:247]: "Docs as code" sounds like a fragment, and readers might not be aware of the concept as it's a phrase that came out of the WTD community. Perhaps better to focus on teams collaborating on this bullet point?

**Review by agjohnson — COMMENTED** [review]
Noted a bunch of copy changes, but it was really helpful to separate the discussion on the layout of the page and come back to discuss the content separately.

File: readthedocs-theme/templates/readthedocs/product.html
Comment (nienn) [reply readthedocs-theme/templates/readthedocs/product.html:9]: Removed this section.

File: content/pages/product.md
Comment (nienn) [reply content/pages/product.md:7]: I removed the _Lorem Ipsum_ and left just the header part in `product.md` and `homepage.md` -  the `homepage.md` file had the same placeholder content as this one.

File: readthedocs-theme/templates/readthedocs/product.html
Comment (nienn) [reply readthedocs-theme/templates/readthedocs/product.html:27]: I believe it's easier for future iterations to have comments as guidance of what was being done in each section, but since we are not leaving comments anywhere it makes better sense to just remove these too.

File: readthedocs-theme/templates/readthedocs/product.html
Comment (nienn) [reply readthedocs-theme/templates/readthedocs/product.html:75]: "Learn about the engineering workflow" is verbose and also too long to fit...

Is launching the website without the sub-product pages even an option? That would force us to revise the entire user flow through the site and probably take at least half the time it would take us to create the sub-product pages.

File: readthedocs-theme/templates/readthedocs/product.html
Comment (nienn) [reply readthedocs-theme/templates/readthedocs/product.html:81]: Removed the header. See previous appearance.

File: readthedocs-theme/templates/readthedocs/product.html
Comment (nienn) [reply readthedocs-theme/templates/readthedocs/product.html:106]: Removed the header. See previous appearance.

File: readthedocs-theme/templates/readthedocs/product.html
Comment (nienn) [reply readthedocs-theme/templates/readthedocs/product.html:92]: We are missing copywriting here to convey the main benefits RTD can bring to a technical writer, but I have no good suggestions.

File: readthedocs-theme/templates/readthedocs/product.html
Comment (nienn) [reply readthedocs-theme/templates/readthedocs/product.html:232]: Good point. I'll remove "automation".

File: readthedocs-theme/templates/readthedocs/product.html
Comment (nienn) [reply readthedocs-theme/templates/readthedocs/product.html:247]: We talked a little bit about this before.
I see what you mean. "Docs as code" is a bit lost sentence at the moment.

Still I would like to emphasize this idea throughout the website as I believe this is what separates Read the Docs from its competitors:

* Read the Docs treats docs as code and, by this, aims to approach the technical writer teams to the developers.
* Competitors use WYSIWYG.

WYSIWYG is much more flashy and calls for more ostentatious design that is easier to attract user attention. We need to emphasize our differences and strengths too.

File: readthedocs-theme/templates/readthedocs/product.html
Comment (agjohnson) [reply readthedocs-theme/templates/readthedocs/product.html]: I would agree that it's help for future authoring purposes to describe the intended future change, but am a little confused by no comments. We do use code comments, however perhaps what hasn't been illustrated is how we try to use code comments?

We generally discard comments that don't provide future code authors with necessary context on the code. We would generally skip code comments that:

* Describe how code operates (`# This is a loop over the project build array`)
* Placeholder comments or code positioning comments
* Commenting out unused code -- we use version control to track old code

In the case, where this block was important to the structure of the design, it might be helpful to keep this in the template. It could also be moved to an issue. A helpful comment would be:

```jinja
{% comment %}
    This is how we will plan on adding illustrations in the future. The content blocks should be broken up by inverted sections:
    <h3 class="ui header inverted"><img class="ui fluid image" src="" /></h3>
{% endcomment %}
```

To illustrate things on the Python side, here's an example -- just the first file I pulled up. The comments here describe to future authors *why* the code is doing what it's doing:
https://github.com/readthedocs/readthedocs.org/blob/master/readthedocs/doc_builder/python_environments.py#L70-L200

Hopefully that helps! This is probably a bit of a code pattern that stems from our Python background and might be a poorly communicated rule because of that.

File: readthedocs-theme/templates/readthedocs/product.html
Comment (agjohnson) [reply readthedocs-theme/templates/readthedocs/product.html]: How much time would you say it would take to author the subproduct pages and finalize copy on the pages?

If it's something we can produce this sprint then that seems fine. However, these pages are all new content, and so I would suspect these will be particularly hard to author.

The product and subproduct pages are all new content, so anything would be better than nothing. I agree reader flow is a problem that we need to have in mind when authoring these pages however. But it's maybe better to view this as an end point rather than a starting point.

File: readthedocs-theme/templates/readthedocs/product.html
Comment (agjohnson) [reply readthedocs-theme/templates/readthedocs/product.html:213]: This is a great point.

"Use your own tools" sort of does that, but I agree that "You don't have to waste your time in a WYSIWG editor" is a separate point entirely.

I don't have any stellar suggestions. Perhaps just expanding this to "Write docs like you write code" or something similar for now?

File: readthedocs-theme/templates/readthedocs/product.html
Comment (nienn) [reply readthedocs-theme/templates/readthedocs/product.html]: Sorry, I miswrote. I meant to say we are not leaving any commented out code, which, in this case, would function as placeholder for future development.

Knowing the patterns in use with more detail helps. Thanks!

In this case I think it's more helpful to create a separate issue so I opened one here #36.
This allowed to better detail the goals for the illustrations that are needed here and will be a better place for the  discussion on the scope of those images.

File: readthedocs-theme/templates/readthedocs/product.html
Comment (nienn) [reply readthedocs-theme/templates/readthedocs/product.html:213]: I opened an  issue related to this topic here #31.

Comment (nienn) [reply readthedocs-theme/templates/readthedocs/product.html:213]: +1 on "Write docs like you write code" or maybe to shorten it a bit "Write docs like you code".

File: readthedocs-theme/templates/readthedocs/product.html
Comment (nienn) [reply readthedocs-theme/templates/readthedocs/product.html]: Ps: We can also refer back to the mockups later, if needed, when adding code back that was already mocked up.

File: readthedocs-theme/templates/readthedocs/product.html
Comment (nienn) [inline readthedocs-theme/templates/readthedocs/product.html:33]: Apply Wikipedia style guide here too:

```suggestion
                  <div class="description">Version control</div>
```

File: readthedocs-theme/templates/readthedocs/product.html
Comment (nienn) [reply readthedocs-theme/templates/readthedocs/product.html]: There where changed to this list introduced by PR #23 witch was being worked in parallel. I ported those changes here with commit https://github.com/readthedocs/site-community/pull/28/commits/092f6408cafe772c38c412c07296a64a5106f737. Please check to see if this feels like a good solution.

File: readthedocs-theme/templates/readthedocs/product.html
Comment (nienn) [reply readthedocs-theme/templates/readthedocs/product.html:220]:
```suggestion
              <span class="ui label basic large"><i class="icon primary hourglass outline"></i> Write docs like you code</span>
```

File: readthedocs-theme/templates/readthedocs/product.html
Comment (nienn) [reply readthedocs-theme/templates/readthedocs/product.html]: I updated  the copy and added links to blog pages related to each workflow: https://docs.readthedocs.io/en/stable/guides/developers.html
https://docs.readthedocs.io/en/stable/guides/authors.html


_commit _

File: readthedocs-theme/templates/readthedocs/product.html
Comment (nienn) [reply readthedocs-theme/templates/readthedocs/product.html]: This section also got some changes from PR #23 that run in parallel. I ported those changes here with commit https://github.com/readthedocs/site-community/pull/28/commits/092f6408cafe772c38c412c07296a64a5106f737.

File: readthedocs-theme/templates/readthedocs/product.html
Comment (nienn) [reply readthedocs-theme/templates/readthedocs/product.html]: Same as above. Added provisory links to blog pages related to each workflow with commit f072830.

For the button copy, maybe we could use the same copy that is used on those new blog pages, for both this button and the one in the next column: "Guides for documentation authors" and "Guides for project administrators"? 

Also, several instances of the `bui `  typo have been removed.

File: readthedocs-theme/templates/readthedocs/product.html
Comment (nienn) [inline readthedocs-theme/templates/readthedocs/product.html:94]:
```suggestion
            <a href="https://docs.readthedocs.io/en/stable/guides/authors.html" class="ui button green fluid">Guides for documentation authors</a>
```

File: readthedocs-theme/templates/readthedocs/product.html
Comment (nienn) [inline readthedocs-theme/templates/readthedocs/product.html:132]:
```suggestion
            <a href="https://docs.readthedocs.io/en/stable/guides/administrators.html" class="ui button green fluid">Guides for project administrators</a>
```

File: readthedocs-theme/templates/readthedocs/product.html
Comment (nienn) [reply readthedocs-theme/templates/readthedocs/product.html]: Change to "Guides for project administrators" as in the blog headings?

_Related to discussion above._

File: readthedocs-theme/templates/readthedocs/product.html
Comment (nienn) [reply readthedocs-theme/templates/readthedocs/product.html]: 👍 I altered the order of some of the list cards.

File: readthedocs-theme/templates/readthedocs/product.html
Comment (nienn) [reply readthedocs-theme/templates/readthedocs/product.html]: This was also ported from #23.

File: readthedocs-theme/templates/readthedocs/product.html
Comment (nienn) [reply readthedocs-theme/templates/readthedocs/product.html]: 👍 Done!

File: readthedocs-theme/templates/readthedocs/product.html
Comment (nienn) [reply readthedocs-theme/templates/readthedocs/product.html]: Done!

File: readthedocs-theme/templates/readthedocs/product.html
Comment (nienn) [reply readthedocs-theme/templates/readthedocs/product.html]: Agree! It's done.

File: readthedocs-theme/templates/readthedocs/product.html
Comment (nienn) [reply readthedocs-theme/templates/readthedocs/product.html]: Done.

**Comment by nienn** [comment]
> Noted a bunch of copy changes, but it was really helpful to separate the discussion on the layout of the page and come back to discuss the content separately.

I absolutely agree! It's really hard to review copy in the same PRs that we review code and project scaffolding and also more prone to create conflicts with other PRs or lost arguments.

Is there anything I can do to promote separate discussions?

I first thought of these latest PR batch as simply porting the `mockup/`s code to the `template/`s logic. We ended up revising a lot more at once and, again, seem to have fallen into the trap of the big PR. This PR is now introducing a lot of new changes but it's still far from reflecting a final polished version of the page, because there are still a lot of elements missing: like links, images, javascript, style adjustments, template logic, etc.

I propose we try to close these PRs as a second mockup and come back to them with separate PRs, each focused on one single objective, to polishing each technical, conceptual, graphic and scaffolding aspects separately.

**Comment by agjohnson** [comment]
> I propose we try to close these PRs as a second mockup and come back to them with separate PRs, each focused on one single objective, to polishing each technical, conceptual, graphic and scaffolding aspects separately.

@nienn That's fair! Would it help to maybe open an issue for each page that have ongoing copy work then? We have been merging development level display changes lately, so what is merged isn't yet complete already. So, it's fine to be merging more development level copy as well then, but I do want to keep up momentum on completing the pages.

I would hold off on tweaking any graphics until we have copy though, there are a few places where graphics are meant to match copy, and the copy could change.

File: readthedocs-theme/templates/readthedocs/product.html
Comment (agjohnson) [reply readthedocs-theme/templates/readthedocs/product.html]: Agreed. Let's come back to this one with more copy review

**Comment by nienn** [comment]
> Would it help to maybe open an issue for each page that have ongoing copy work then?

Yes! This absolutely helps. We might become a bit overwhelmed with these issues soon thought. Maybe we can start adding issues to milestones and/or add appropriate labels to keep track of the ones that are required before any release. Also maybe start adding labels to distinguish between `development`, `new feature` and `bug` issues.

> I would hold off on tweaking any graphics until we have copy though, there are a few places where graphics are meant to match copy, and the copy could change.

Yes, I agree. I would only add that its important to know where there will be a graphic and the general shape/concept of that graphic, as that will alter the copy too. Both should live together.

File: readthedocs-theme/templates/readthedocs/product.html
Comment (agjohnson) [reply readthedocs-theme/templates/readthedocs/product.html]: That's a great compromise for now. I do want to keep these on our radar, but in the interest in keeping the scope of your work down, we can come back to the subproduct pages next.

I opened #43 to track this

File: readthedocs-theme/templates/readthedocs/product.html
Comment (agjohnson) [reply readthedocs-theme/templates/readthedocs/product.html]: Yeah, that sounds better. If we don't like how it looks we can tweak later.

File: readthedocs-theme/templates/readthedocs/product.html
Comment (agjohnson) [reply readthedocs-theme/templates/readthedocs/product.html]: Yeah, that copy makes sense

**Comment by agjohnson** [comment]
> Maybe we can start adding issues to milestones

Yeah, I was going to ask if milestones would be helpful. We are accumulating issues and prioritization would help. We can also move things to the roadmap view in Trello if you'd prefer. I'm open to either.

Feel free to apply any of the labels that you think make sense. We already use some of the labels you mention, if you haven't seen it yet, our issue triaging guidelines are here: https://dev.readthedocs.io/en/latest/issue-labels.html

**Comment by agjohnson** [comment]
I updated the base to `main` but this branch could use a rebase or merge of main, to isolate the pr changes here

**Comment by nienn** [comment]
I'm going to commit these two suggestions - https://github.com/readthedocs/site-community/pull/28#discussion_r792075215 and https://github.com/readthedocs/site-community/pull/28#discussion_r792075564 - and rebase.

**Review by agjohnson — APPROVED** [review]
This also looks ready to merge! If there are improvements to the copy etc, we'll address these with some iteration.

**Pull request merged**
